ns3_adapter/src/ns3_adapter.cpp

```diff
@@ -267,29 +279,29 @@ void NS3Adapter::pre_spin()
             {
                 std::lock_guard<std::mutex> lock(cfg_mutex_);
             }
-            ROS_INFO_STREAM("Attempting to connect to NS3");
+            ROS_DEBUG("Attempting to connect to NS3");
             boost::system::error_code ec;
            // ROS_INFO("Connecting to %s:%u", cfg.dsrc_address.c_str(), cfg.dsrc_listening_port);
-            ROS_INFO("Local port: %u", ns3_broadcasting_port_);
+            ROS_DEBUG("Remote broadcasting port: %u", ns3_broadcasting_port_);
+            ROS_DEBUG("Remote registration port: %u", ns3_registration_port_);
+
             try {
-                if (!ns3_client_.connect(ns3_address_, ns3_broadcasting_port_,
-                                          ns3_listening_port_, ec))
+                if (!ns3_client_.connect_registration_and_broadcasting(ns3_address_, ns3_broadcasting_port_, ns3_registration_port_,
+                                          ns3_v2x_listening_port_, ns3_time_listening_port_, ec))
                 {
                     ROS_WARN_STREAM("Failed to connect, err: " << ec.message());
                 }
             }catch(std::exception e)
             {
                 ROS_ERROR_STREAM("Exception connecting to CARMA Ambassador: " << e.what() << " error_code: " << ec.message());
             }
-
-
             connecting_ = false;
         }));
     }
 
-    ns3_reg_client_.connect(ns3_address_, ns3_registration_port_);
-    std::string handshake_msg = compose_handshake_msg(vehicle_id_, role_id_, std::to_string(ns3_listening_port_), host_ip_);
-    ROS_WARN_STREAM("handshake_msg: " << handshake_msg);
+    //ns3_client_.connect_registration(ns3_address_, ns3_registration_port_);
+    std::string handshake_msg = compose_handshake_msg(vehicle_id_, role_id_, std::to_string(ns3_v2x_listening_port_), std::to_string(ns3_time_listening_port_), host_ip_);
```

Comment:
Is there a reason why we are converting ports to strings. we could just edit the compose_handshake_msg to take shorts or what ever they are?

Reply:
Yeah we can do that. It is becoming JSON string in the end anyways, so I just left it as it is.

Reply:
Ints and string are represented differently in json. I think most parsers are smart enough to get around this but technically what infrastructure is expecting is 
```json
{
"vehicle_id":"some_id",
"carla_role_id":"some_id",
"v2xMessageRxPort":1233,
"timeSyncRxPort":3432,
"host_ip":"123.3.3.123"
}
```
I know the field names are wrong but the idea is numbers should not have quotes in json or else they will be interpreted as stringw.